public/scripts/tutorials.js

```diff
@@ -102,36 +102,49 @@ define(["jquery", "/external/make-api.js", "/external/requestAnimationFrameShim.
             return;
           } else if (data.action === "highlight") {
             var top = codeMirror.getScrollInfo().top,
-                clientHeight = codeMirror.getScrollInfo().clientHeight,
-                viewportTop = codeMirror.lineAtHeight(top + 10, "local"),
-                viewportBottom = codeMirror.lineAtHeight(top + clientHeight, "local"),
-                mark;
-
-            if (lineTo < viewportTop) {
-              $(".tutorial-pointer").text(lineString).show();
-            } else if (lineFrom > viewportBottom) {
-              $(".tutorial-pointer").text(lineString).addClass("down").show();
-            }
+                clientHeight = codeMirror.getScrollInfo().clientHeight;
+
+            codeMirror.lineAtHeight(top + 10, "local", function(viewportTop) {
```

Comment:
Comment on this change, since it's coupled to our proxy but looks like codemirror.